### ISSUE TITLE
Crash in Physics project, Kinematic.entity when generating balls

### ISSUE THREAD

**Issue by geekrelief** [OurMachinery/themachinery-public#708]
I speed up the simulation time to have it trigger more quickly. The simulation seems to run fine as long as you don't generate a ball.  Generating more balls increase the likelihood of a crash.

https://user-images.githubusercontent.com/27040/128781770-e03bbaf4-a84f-4c6a-9843-28e62f8a7df9.mp4

**Comment by khskarl** [comment]
Thanks for reporting! This may be same problem I have been getting, but it was harder to reproduce in my case as it only happened sometimes when there are lots of collisions in a ragdoll.

**Comment by karll-ourmachinery** [comment]
Thanks for reporting! This may be same problem I have been getting, but it was harder to reproduce in my case as it only happened sometimes when there are lots of collisions in a ragdoll.

**Comment by niklas-ourmachinery** [comment]
Reproduced and investigating.

**Comment by niklas-ourmachinery** [comment]
Reproduced the crash with a PhysX debug build. The crash happens in the default assert handler in PhysX, triggered by `PXASSERT(mManager)` in `Sc::ShapeInteraction::destroyManager()`.

Full call stack:

~~~
JobSystemCpuDispatcher::run_task()
physx::Cm::Task::run()
physx::Cm::DelegateTask::runInternal()
physx::Sc::Scene::updateCCDSinglePassStage3()
physx::Sc::Scene::finishBroadPhaseStage2()
physx::Sc::ShapeInteraction::destroyManager()
DefaultAssertHandler::operator()
~~~

**Comment by niklas-ourmachinery** [comment]
Running with PhysX default CPU dispatcher instead of our job-based one produces the same crash, so it's not caused by interaction with the job system.

**Comment by niklas-ourmachinery** [comment]
I can't reproduce the behavior with the latest PhysX version... will update our internal version.

**Comment by niklas-ourmachinery** [comment]
Fixed for 2021.8 by updating the PhysX version.
